src/ui_parts/viewport.gd

```diff
@@ -100,3 +100,24 @@ func adjust_view() -> void:
 func _on_size_changed() -> void:
 	if is_node_ready():
 		adjust_view()
+
+func wrap_mouse(relative: Vector2) -> Vector2:
+	var view_rect:Rect2 = get_visible_rect().grow(-1.0)
+	var mouse_pos: Vector2 = get_mouse_position()
+	
+	var warp_margin := view_rect.size * 0.5
+	
+	var relative_sign := Vector2i(
+		1 if relative.x >= 0.0 else -1, 
+		1 if relative.y >= 0.0 else -1)
```

Comment:
Are you aware of Vector2.sign()? Looks like a more appropriate thing to use here.

Reply:
I was not aware